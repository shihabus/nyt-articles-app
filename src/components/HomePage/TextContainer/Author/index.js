import React from 'react'
import { View, Text,StyleSheet } from 'react-native'
import PropTypes from 'prop-types';

const styles=StyleSheet.create({
    greyText: {
        fontSize: 14,
        color: "gray"
      },
})

const Author = ({byline}) => {
    return (
        <View>
          <Text style={styles.greyText} testID='Author Text'>{byline}</Text>
        </View>
    )
}

Author.propTypes={
    byline:PropTypes.string
}



export default Author
