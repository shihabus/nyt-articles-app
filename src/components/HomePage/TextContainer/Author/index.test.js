/* eslint-disable no-undef */
// /* eslint-disable import/named */
// /* eslint-disable no-undef */
// /**
//  * @jest-environment jsdom
//  */

import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Author from "./index";
import { checkProps,findComponent } from "../../../../../testUtils";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const setUp = (props = {}) => {
  const wrapper = shallow(<Author {...props} />);
  return wrapper;
};

describe("Author", () => {
  let wrapper;
  beforeEach(() => {
    const props = {
      byline: "Test"
    };
    wrapper = setUp(props);
  });

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot("Author");
  });

  it("Checked props", () => {
    const expectedProps = {
      byline: "Test"
    };
    const propsErr = checkProps(Author, expectedProps);
    expect(propsErr).toBeUndefined();
  });

  it('should render child',()=>{
    expect(findComponent(wrapper,"Author Text").length).toEqual(1)
  })
});
