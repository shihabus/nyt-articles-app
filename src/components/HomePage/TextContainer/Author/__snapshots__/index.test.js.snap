// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Snapshot: Author 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "gray",
        "fontSize": 14,
      }
    }
    testID="Author Text"
  >
    Test
  </Text>
</View>
`;
